pkg/kf/sources/util.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sources
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/google/kf/pkg/apis/kf/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// SourceStatus gets the status of the given source.
+// Complete will be set to true if the build has completed (or doesn't exist).
+// Error will be set if the build completed with an error (or doesn't exist).
+// A successful result is one that completed and error is nil.
+func SourceStatus(source v1alpha1.Source) (finished bool, err error) {
+	condition := source.Status.GetCondition(v1alpha1.SourceConditionSucceeded)
+	if condition == nil {
+		// no success condition means the build hasn't propigated yet
+		return false, nil
+	}
+
+	switch condition.Status {
+	case corev1.ConditionTrue: // the build was a success
```

Comment:
nit: these comments on ConditionTrue and False are not necessary. However, the one for default is nice.